improver/calibration/rainforest_calibration.py

```diff
@@ -0,0 +1,170 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""RainForests calibration Plugins."""
+
+import warnings
+
+import numpy as np
+from iris.cube import Cube, CubeList
+
+from improver import BasePlugin, PostProcessingPlugin
+
+# Passed to choose_set_of_percentiles to set of evenly spaced percentiles
+DEFAULT_ERROR_PERCENTILES_COUNT = 19
+DEFAULT_OUTPUT_REALIZATIONS_COUNT = 100
+
+
+class TrainRainForestsTreeModels(BasePlugin):
+    """Class to train tree models for use in RainForests calibration."""
+
+
+class ApplyRainForestsCalibration(PostProcessingPlugin):
+    """Class to calibrate input forecast given a series of RainForests tree models."""
+
+    def __init__(self, model_config_dict: dict, nthreads: int):
+        """Initialise the tree model variables used in the application of RainForests
+        Calibration.
+
+        Args:
+            model_config_dict:
+                Dictionary containing Rainforests model configuration variables.
+            nthreads:
+                Number of threads to use when initialising tree-model objects.
+
+        Dictionary is of format::
+
+            {
+                "-50.0" : {
+                    "lightgbm_model" : "<path_to_lightgbm_model_object>",
+                    "treelite_model" : "<path_to_treelite_model_object>"
+                },
+                "-25.0" : {
+                    "lightgbm_model" : "<path_to_lightgbm_model_object>",
+                    "treelite_model" : "<path_to_treelite_model_object>"
+                },
+                ...,
+                "50.0" : {
+                    "lightgbm_model" : "<path_to_lightgbm_model_object>",
+                    "treelite_model" : "<path_to_treelite_model_object>"
+                }
+            }
+
+        The keys specify the error threshold value, while the associated values
+        are the path to the corresponding tree-model objects for that threshold.
+
+        Treelite Predictors are returned if treelite_runtime is an installed dependency and
+        an associated path has been supplied for all thresholds, otherwise lightgbm the
+        Boosters are returned.
+
+        Returns:
+            - numpy array containing the error threshold values.
+            - list of tree-model objects to be used in calibration process.
+        """
+        import importlib
+
+        from lightgbm import Booster
+
+        if importlib.util.find_spec("treelite") is not None:
+            from treelite_runtime import Predictor
+
+            self.treelite_enabled = True
+        else:
+            warnings.warn(
+                "Module treelite_runtime unavailable. Defaulting to using lightgbm Boosters."
+            )
+            self.treelite_enabled = False
```

Comment:
Suggest using an "ask for forgiveness rather than permission" approach here.
```suggestion
        try:
            from treelite_runtime import Predictor
            self.treelite_enabled = True
        except ModuleNotFoundError:
            warnings.warn(
                "Module treelite_runtime unavailable. Defaulting to using lightgbm Boosters."
            )
            self.treelite_enabled = False
```

Reply:
There's another similar usage in the unit test code.

Reply:
Good suggestion! I've updated both instances as suggested.